odl/discr/grid.py

```diff
@@ -451,54 +532,85 @@ def __contains__(self, other):
+        >>> rg_sub = uniform_grid([-1, 2], [0, 4], (2, 2))
+        >>> rg_sub.coord_vectors
+        (array([-1.,  0.]), array([ 2.,  4.]))
+        >>> rg_sub.is_subgrid(rg)
         True
-        >>> g_sub = TensorGrid([0.1], [-1.05, 2.1])
-        >>> g_sub.is_subgrid(g)
+
+        Fuzzy check is also possible. Note that the tolerance still
+        applies to the coordinate vectors.
+
+        >>> rg_sub = uniform_grid([-1.015, 2], [0, 3.99], (2, 2))
+        >>> rg_sub.is_subgrid(rg, atol=0.01)
         False
-        >>> g_sub.is_subgrid(g, atol=0.15)
+        >>> rg_sub.is_subgrid(rg, atol=0.02)
         True
         """
         # Optimization for some common cases
         if other is self:
             return True
-        if not (isinstance(other, TensorGrid) and
+        if not (isinstance(other, RectGrid) and
                 np.all(self.shape <= other.shape) and
                 np.all(self.min_pt >= other.min_pt - atol) and
                 np.all(self.max_pt <= other.max_pt + atol)):
             return False
 
-        # Array version of the fuzzy subgrid test, about 3 times faster
-        # than the loop version.
-        for vec_o, vec_s in zip(other.coord_vectors, self.coord_vectors):
-            # Create array of differences of all entries in vec_o and vec_s.
-            # If there is no almost zero entry in each row, return False.
-            vec_o_mg, vec_s_mg = sparse_meshgrid(vec_o, vec_s)
-            if not np.all(np.any(np.abs(vec_s_mg - vec_o_mg) <= atol, axis=0)):
-                return False
+        if self.is_uniform and other.is_uniform:
+            # For uniform grids, it suffices to show that min_pt, max_pt
+            # and g[1,...,1] are contained in the other grid. For axes
+            # with less than 2 points, this reduces to min_pt and max_pt,
+            # and the corresponding indices in the other check point are
+            # set to 0.
+            minmax_contained = (
+                other.approx_contains(self.min_pt, atol=atol) and
+                other.approx_contains(self.max_pt, atol=atol))
+            check_idx = np.zeros(self.ndim, dtype=int)
+            check_idx[np.array(self.shape) >= 3] = 1
+            checkpt_contained = other.approx_contains(self[tuple(check_idx)],
+                                                      atol=atol)
+            return minmax_contained and checkpt_contained
 
-        return True
+        else:
+            # Array version of the fuzzy subgrid test, about 3 times faster
+            # than the loop version.
+            for vec_o, vec_s in zip(other.coord_vectors, self.coord_vectors):
+                # Create array of differences of all entries in vec_o and
+                # vec_s. If there is no almost zero entry in each row,
+                # return False.
+                vec_o_mg, vec_s_mg = sparse_meshgrid(vec_o, vec_s)
+                if not np.all(np.any(np.abs(vec_s_mg - vec_o_mg) <= atol,
```

Comment:
No, `allclose` is not suitable since the two vectors are broadcast and form a matrix (like a distance matrix). Then it is checked that in each column there is at least one entry close enough to zero.
AFAICS the only thing I can do is replace the boolean abs expression with `np.isclose`.